src/Client/ClientBase.cpp

```diff
@@ -1821,6 +1821,11 @@ void ClientBase::processParsedSingleQuery(const String & full_query, const Strin
     client_exception.reset();
     server_exception.reset();
 
+    Settings settings = client_context->getSettingsCopy();
+    settings.script_line_number = settings.script_line_number + 1;
```

Comment:
This is inefficient and also incorrect. It counts queries, not lines.

Reply:
Yes, I agree setting the copy of settings each query, is inefficient, I will try to find some different place to put it.

Reply:
And ofc, make it count each line, some hint will be apprieciated. But i will try to resolve it on my hand.